apps/studio/stores/RootStore.ts

```diff
@@ -35,16 +31,8 @@ export class RootStore implements IRootStore {
    * This method will also trigger project detail loading when it's not available
    */
   setProject(project: Project) {
-    if (this.selectedProjectRef === project.ref) return
```

Comment:
If the connection string changed but the project ref doesn't, we skip the update. This is the real fix.